### PR TITLE
The CLI can now resolve shared fragments

### PR DESCRIPTION
**Pull request by minasmart**
The pattern that's used is looking in the same directory as the graphql file for a file named the same as the fragment in document.

Example:

```graphql
query {
  node(id: "abc123") {
    ...ProductFragment
  }
}
```

Will search for a document passed in that lives in the same directory and has the name `ProductFragment.graphl`.

This PR also includes the functions required to search for undefined fragments and find the names of matching fragment files for any document. These functions will likely be useful in any plugins building on this functionality.

### CODE REVIEW

File: src/cli-controller.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ import {
 } from './index';
 
 function normalizePath(filename) {
```
Comment (swalkinshaw) [inline src/cli-controller.js]: Can this whole function just be replaced with `path.resolve`?

ref: https://nodejs.org/api/path.html#path_path_resolve_paths

Comment (minasmart) [inline src/cli-controller.js]: ooooohhhh... fancy.

File: src/fragment-files-for-document.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import {dirname, join} from 'path';
+import findUndefinedFragments from './find-undefined-fragments';
+
+export default function fragmentFilesForDocument(documentPath, document) {
+  const undefindeFragments = findUndefinedFragments(document);
```
Comment (swalkinshaw) [inline src/fragment-files-for-document.js]: typo on `undefindeFragments`

File: test/find-undefined-fragments-test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        ...productFragmentPartOne
+        ...productFragmentPartTwo
+      }
+
+      fragment productFragmentPartOne on Product {
+        name
+      }
+    `);
+
+    assert.deepEqual(undefindedFragments, [
```
Comment (swalkinshaw) [inline test/find-undefined-fragments-test.js]: here too

File: src/cli.js [hunk 4]
```diff
@@ -3,30 +3,64 @@ import {relative, basename, dirname, join} from 'path';
...
+    const fragmentFiles = fragmentFilesForDocument(document.path, document.body);
+
+    if (fragmentFiles.length) {
+      const fragments = documentCode.filter((possibleFragment) => {
+        return fragmentFiles.includes(possibleFragment.path);
+      });
+
+      fragments.forEach((fragment) => {
+        fragment.isFragment = true;
+      });
```
Comment (swalkinshaw) [inline src/cli.js]: Only part I found weird was returning documents here but also relying on the side effect of `isFragment` being set. Not sure if there's a better way around it though.

**Pull request merged**